tutorials/notebook.md

```diff
@@ -0,0 +1,578 @@
+dependency set. Edit the new `package.json` file in your favorite editor and add
+a new dependency set containing only the `tupelo-client` library:
+
+In file `notebook/package.json`:
+```json
+{
+  "name": "notebook",
+  ...
+  "dependencies": {
+      "tupelo-client": "^0.0.2-alpha1"
+  },
+}
+```
+
+Next, run `npm install` from the application directory to install the dependency
+to your local cache.
+
+After installing the `tupelo-client` dependency, make a new file called
+`index.js` with your favorite editor to hold all the application code. At the
+top of that file, require the Tupelo client library so we can use it later.
+
+In file `notebook/index.js`:
+```javascript
+const tupelo = require('tupelo-client');
+```
+
+## Creating a New Notebook
+Our application will organize notes in _notebooks_, and we'll store the data
+associated with a particular notebook in its own, unique chain tree. Before we
+can start saving notes, we need to register a wallet at the RPC server to store
+the keys used to access the chain tree, create a new key, and create the chain
+tree we'll use to store our notes. Let's build a `createNotebook()` function,
+step by step, to do all those things.
+
+### Connecting to the RPC Server
+The Tupelo RPC server listens for connections on a specific port of the host its
+running on after it starts. Since we did not specify the port [when we started
+the server](#running-the-tupelo-rpc-server), our server is listening on "50051",
+which is the default port. We will also need to supply a set of wallet
+credentials to establish every connection to the RPC server. The credentials are
+in an object with `walletName` and `passPhrase` keys.
+
+Let's make a `connect` function that accepts wallet credential objects and
+establishes a connection to the running development server.
+
+In file `notebook/index.js`:
+```javascript
+function connect(creds) {
+  return tupelo.connect('localhost:50051', creds);
+}
+```
+
+Next, let's start building the `createNotebook()` function by establishing a
+connection to the RPC server
+
+In file `notebook/index.js`:
+```javascript
+function createNotebook(creds) {
+  let client = connect(creds);
+}
+```
+
+### Registering a New Wallet
+In order to create and manipulate a chain tree in Tupelo, you must first create
+or control the key associated with that chain tree. The RPC server manages those
+keys in different wallets, but the client has to create the wallet and set its
+credentials first. The `register()` method of the tupelo client returns a
+promise that, when fulfilled, either contains the result of creating a new
+wallet with the client's credentials, or an error if the wallet already
+exists or there's another error during creation. Let's extend our
+`createNotebook()` function to register a new wallet.
+
+In file `notebook/index.js`:
+```javascript
+function createNotebook(creds) {
+  let client = connect(creds);
+
+  // ---- Add the code below ----
```

Comment:
Rather than comments here and below, I've seen some walkthroughs leverage diff highlighting to make the new code standout, for example:

```diff
function createNotebook(creds) {
  let client = connect(creds);

+  client.register()
+    .then(function(registerResult){
+      console.log("Success!");
+    }, function(err) {
+      console.log("Error registering wallet.");
+      console.log(err);
+    });
}
```

Trade off is you obviously lose the syntax highlighting in this view, which imho is slightly less readable if someone just browsing and not coding through this.

Reply:
Sorry I missed this comment until now. I think i like the comments over diff formatting because of the syntax highlighting issue you mentioned, and because putting things in as a diff makes it harder to copy and paste.